ci/cscs-ci.yml

```diff
@@ -1,185 +1,135 @@
-build_py310_image_aarch64:
-  extends: .build_image_aarch64
-  needs: [build_py310_baseimage_aarch64]
-  variables:
-    <<: *py310
-
+# .build_cscs_amd_rocm:
+#   extends:
+#     - .container-builder-cscs-zen2
+#     - .build_common
+#     - .build_extra_rocm
 
-.test_helper:
+# -- Test stage --
+.test_common:
   stage: test
-  image: $CSCS_REGISTRY_PATH/public/$ARCH/gt4py/gt4py-ci:$CI_COMMIT_SHA-$PYVERSION
-  script:
-    - cd /gt4py.src
-    - echo "CI_COMMIT_BRANCH=${CI_COMMIT_BRANCH}"
-    - export GITHUB_PR="${CI_COMMIT_BRANCH#__CSCSCI__pr}"
-    - echo "GITHUB_PR=${GITHUB_PR}"
-    - export CI_NOX_RUN_ONLY_IF_CHANGED_FROM="gh/$(curl https://api.github.com/repos/gridtools/gt4py/pulls/${GITHUB_PR} 2>/dev/null | jq -r '.base.ref')"
-    - echo "CI_NOX_RUN_ONLY_IF_CHANGED_FROM=${CI_NOX_RUN_ONLY_IF_CHANGED_FROM}"
-    - NOX_SESSION_ARGS="${VARIANT:+($VARIANT}${SUBVARIANT:+, $SUBVARIANT}${DETAIL:+, $DETAIL}${VARIANT:+)}"
-    - nox -s "test_$SUBPACKAGE-${PYVERSION:0:4}$NOX_SESSION_ARGS"
+  image: ${GT4PY_IMAGE_PREFIX}-${PY_VERSION}
   variables:
     CSCS_CUDA_MPS: 1
     SLURM_JOB_NUM_NODES: 1
-    SLURM_TIMELIMIT: 5
-    PYENV_VERSION: $PYVERSION
-    VIRTUALENV_SYSTEM_SITE_PACKAGES: 1
-    CI_NOX_VERBOSE: 1
-# .test_helper_x86_64:
-#   extends: [.container-runner-daint-gpu, .test_helper]
-#   parallel:
-#     matrix:
-#     - SUBPACKAGE: [cartesian, storage]
-#       VARIANT: [-internal, -dace]
-#       SUBVARIANT: [-cuda11x, -cpu]
-#     - SUBPACKAGE: eve
-#     - SUBPACKAGE: next
-#       VARIANT: [-nomesh, -atlas]
-#       SUBVARIANT: [-cuda11x, -cpu]
-.test_helper_aarch64:
-  extends: [.container-runner-santis-gh200, .test_helper]
+    SLURM_TIMELIMIT: 4
   parallel:
     matrix:
       - SUBPACKAGE: [cartesian]
-        VARIANT: ['internal', 'dace']
-        SUBVARIANT: ['cuda12', 'cpu']
+        VARIANT: ["internal", "dace"]
+        SUBVARIANT: ["cuda12", "cpu"]
+        PY_VERSION: *python_versions
       - SUBPACKAGE: eve
+        PY_VERSION: *python_versions
       - SUBPACKAGE: next
-        VARIANT: ['internal', 'dace']
-        SUBVARIANT: ['cuda12', 'cpu']
-        DETAIL: ['nomesh', 'atlas']
+        VARIANT: ["internal", "dace"]
+        SUBVARIANT: ["cuda12", "cpu"]
+        DETAIL: ["nomesh", "atlas"]
+        PY_VERSION: *python_versions
       - SUBPACKAGE: [storage]
-        VARIANT: ['cuda12', 'cpu']
+        VARIANT: ["cuda12", "cpu"]
+        PY_VERSION: *python_versions
   rules:
     - if: $SUBPACKAGE == 'next' && $VARIANT == 'dace' && $DETAIL == 'nomesh'
       variables:
         # TODO: investigate why the dace tests seem to hang with multiple jobs
         GT4PY_BUILD_JOBS: 1
-        SLURM_TIMELIMIT: '00:15:00'
+        SLURM_TIMELIMIT: "00:15:00"
     - when: on_success
+  script:
+    # Since the image does not contain the repo, we need to clone it before running the tests
+    #- mkdir -p "${WORKDIR}/gt4py" && git clone --depth 1 --revision "${CI_COMMIT_SHA}" "${CSCS_CI_ORIG_CLONE_URL}" "${WORKDIR}/gt4py"
```

Comment:
this didn't work? do I read correctly that you do clone default branch + fetch current branch now?

Reply:
The commented version works, but requires git >= 2.49, which is not available in the docker image (and I miserably failed when I tied to install it in the dockerfile). The current line is the best thing I could find to download the minimal amount of information from the git repo that works with older git versions.